libnmstate/validator.py

```diff
@@ -400,6 +402,18 @@ def _validate_linux_bond(original_desired_state, current_state):
             _assert_bond_config_is_legal(iface_state, original_iface_state)
 
 
+def _validate_vlan(desired_state, current_state):
+    for iface_state in desired_state.interfaces.values():
+        if iface_state[Interface.TYPE] == InterfaceType.VLAN:
+            base_iface_name = iface_state.get(VLAN.CONFIG_SUBTREE).get(VLAN.BASE_IFACE)
+            base_iface_cur_state = current_state.interfaces.get(base_iface_name)
+            if base_iface_cur_state[Interface.TYPE] == InterfaceType.ETHERNET:
+                base_iface_mtu = base_iface_cur_state.get(Interface.MTU)
+                vlan_mtu = iface_state.get(Interface.MTU)
+                if vlan_mtu > base_iface_mtu:
+                    raise ValueError("MTU of VLAN can not be more than MTU of base Interface")
```

Comment:
It would be nice if the error shows up the information, something like: "vlan1 mtu (number) cannot be greater than its base interface eth1 mtu (number)". It should be easy because you already have all the information stored in variables.